audtorch/datasets/speech_commands.py

```diff
@@ -0,0 +1,118 @@
+import os
+import random
+
+from .utils import (download_url, extract_archive, safe_path)
+from .base import AudioDataset
+from os.path import join
+
+__doctest_skip__ = ['*']
+
+
+class SpeechCommands(AudioDataset):
+    r"""Data set of spoken words designed for keyword spotting tasks.
+
+    Speech Commands V2 publicly available from Google:
+    http://download.tensorflow.org/data/speech_commands_v0.02.tar.gz
+
+    License: CC BY 4.0
+
+    Args:
+        root (str): root directory of data set, where the CSV files are
+            located, e.g. `/data/speech_commands_v0.02`
+        train (bool, optional): Partition the dataset into the training set.
+            `False` returns the test split. Default: False.
+        download (bool, optional): Download the dataset to `root` if it's not
+            already available. Default: False
+        include (list of str, optional): list of categories to include as
+            commands. If `None` all categories are included. Default:
+            `['yes','no','up','down','left','right','on','off','stop','go']`
+        silence (bool, optional): include a 'silence' class composed of
+            background noise. (Note: use randomcrop when training)
```

Comment:
Regarding random crop and the different length of the background noise:

1. If we stay with the current behavior, please change the note to ```(Note: use ``transforms.RandomCrop(16000)`` when training to ensure the same length for silence as for the other classes)```
2. Alternatively we could directly integrate `RandomCrop` with silence as part of the data set, but maybe this is not wanted as the original data is 1min in duration.

As I'm not familiar with the keyword spotting community, which one do you think would fit better?

Reply:
Hi, thanks for the feedback - I was unsure of this as well. 

> One could think about replacing train=True and selecting n_samples * 0.9 by a user option to specify more directly the amount of silence, but i think for now we can just go with your solution.

I've done it this way just to comply with the train/valid/test split in the original paper, but I'm not sure how often experiments adhere to this. I agree that splitting it that way would make it easier. 

> As I'm not familiar with the keyword spotting community, which one do you think would fit better?

Me neither - I'm mostly in the sound event detection community :-). I think ease of use should be prioritised as the `silence` class is not that interesting to classify / detect.

**Proposal 2** would definitely be easier to get up and running in the long run - users simply wouldn't need to worry about the `silence` class as it's provided "as is". I think that this route would be fine as long as there's a comment explaining what's going on. The current behaviour is more hands off but could lead to some confusion & requires a bit more extra setup, as well as the fact that `RandomCrop` needs to be applied to the entire dataset (This is common for speech commands but some people may not want to do this).

There is a design decision question however: seeing as we integrate the random crop into the silence class, is it possible to exclude the samples from a `transform` applied to the dataset? I'm just thinking about the result in this paper - https://arxiv.org/abs/1904.08779, where e.g. one of the augmentations is that certain frequency bands are dropped. I'm unsure if it would be desirable to augment the `silence` class in this case. What do you think?

Reply:
OK, in summary: it would convenient to integrate random cropping, but we have to warn people about it and provide the possibility to disable it.

How about this proposal:
Add a `same_length=True` argument to the data set. If it is `True` then we apply `transforms.RandomCrop(16000)` to `silence` samples only and state that in the docstring. If it is `False` we just return the 1min signals as at the moment.

Reply:
That sounds sane to me. Will implement today.

Reply:
Resolved with https://github.com/audeering/audtorch/pull/49/commits/d3a63254541e6cf6eeb4168f3236fe14fff54dfd

Sorry it took some time - attention got dragged to other work. Please let me know if there are any further changes required.